receiver/redisreceiver/integration_test.go

```diff
@@ -46,5 +47,75 @@ func TestIntegration(t *testing.T) {
 				return redisPort
 			}),
 		),
+		scraperinttest.WithExpectedFile(filepath.Join("testdata", "integration", "expected-old.yaml")),
+	).Run(t)
+}
+
+func TestLatestIntegration(t *testing.T) {
+	scraperinttest.NewIntegrationTest(
+		NewFactory(),
+		scraperinttest.WithContainerRequest(
+			testcontainers.ContainerRequest{
+				Image:        "redis:7.2.3",
+				ExposedPorts: []string{redisPort},
+				WaitingFor:   wait.ForListeningPort(redisPort),
+			}),
+		scraperinttest.WithCustomConfig(
+			func(t *testing.T, cfg component.Config, ci *scraperinttest.ContainerInfo) {
+				rCfg := cfg.(*Config)
+				rCfg.Endpoint = fmt.Sprintf("%s:%s", ci.Host(t), ci.MappedPort(t, redisPort))
+			}),
+		scraperinttest.WithCompareOptions(
+			pmetrictest.IgnoreMetricValues(),
+			pmetrictest.IgnoreMetricDataPointsOrder(),
+			pmetrictest.IgnoreStartTimestamp(),
+			pmetrictest.IgnoreTimestamp(),
+			pmetrictest.ChangeResourceAttributeValue("server.address", func(_ string) string {
+				return "localhost"
+			}),
+			pmetrictest.ChangeResourceAttributeValue("server.port", func(_ string) string {
+				return redisPort
+			}),
+		),
+		scraperinttest.WithExpectedFile(filepath.Join("testdata", "integration", "expected-latest.yaml")),
+	).Run(t)
+}
+func TestClusterIntegration(t *testing.T) {
```

Comment:
Does this test cover all the metrics validated by the previous test? If so, we can keep only the cluster test, right? Maybe couple of them: cluster with v6 and cluster with v7...

BTW it's a nice test!